ddtrace/context.py

```diff
@@ -50,13 +56,28 @@ def get_current_span(self):
         with self._lock:
             return self._current_span
 
+    def _set_current_span(self, span):
```

Comment:
Because the class is meant to be thread-safe, whatever API we provide, it should use the internal `lock` for any change even if it's an internal API.

Reply:
That was just not to dupe code between `add_span` and `close_span` which themselves set the lock.
If we think that's too dangerous even if internal, we can inline it in these two functions.